### PR TITLE
Delta Lake +Unity Catalog exporter: Delta Lake table metadata

### LINKED ISSUE #7302: Unity catalog exporter: Exported tables should retain original table field documentation
databricks users can add comments to tables in Unity Catalog and to their columns for documentation purposes. 
[Table-level comments](https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-comment.html) can be added after table creation, but [column-level comments](https://docs.delta.io/1.1.0/api/java/io/delta/tables/DeltaTableBuilder.html#comment-java.lang.String-) can be added on table creation. 

Currently, table and column comments don't survive the export process to Unity Catalog. 

### Requirements 

* Retain Delta Lake tables comments on their export to Unity 
* Retain Delta Lake columns comments on their export to Unity

### PR DESCRIPTION
**Pull request by Jonathan-Rosenberg**
Closes #7302

## Change Description

* When returning the Delta Log, also return the table's metadata. 
* Unity Catalog: if the external Delta Lake table has a description, create it in Unity with it as a comment.

FYI @talSofer

### CODE REVIEW

File: pkg/actions/lua/databricks/client.go [hunk 1]
```diff
@@ -8,6 +8,8 @@ import (
 	"regexp"
 	"strings"
 
+	luautil "github.com/treeverse/lakefs/pkg/actions/lua/util"
+
```
Comment (N-o-Z) [inline pkg/actions/lua/databricks/client.go:12]:
```suggestion
	luautil "github.com/treeverse/lakefs/pkg/actions/lua/util"
```

File: pkg/actions/lua/formats/delta.go [hunk 2]
```diff
@@ -7,6 +7,8 @@ import (
 	"net/url"
 	"regexp"
 
+	"github.com/csimplestring/delta-go/action"
+
```
Comment (N-o-Z) [inline pkg/actions/lua/formats/delta.go:11]:
```suggestion
	"github.com/csimplestring/delta-go/action"
```

File: docs/howto/hooks/lua.md [hunk 3]
```diff
@@ -373,8 +373,17 @@ Creates a new Delta Lake client used to interact with the lakeFS server.
 ### `formats/delta_client.get_table(repository_id, reference_id, prefix)`
 
 Returns a representation of a Delta Lake table under the given repository, reference, and prefix.
-The format of the response is a table `{number, {string}}` where `number` is a version in the Delta Log, and the mapped 
-`{string}` table (list) contains JSON strings of the different Delta Lake log operations listed in the mapped version entry. 
+The format of the response is two tables: 
+- the first is a table of the format `{number, {string}}` where `number` is a version in the Delta Log, and the mapped `{string}` 
+table (list) contains JSON strings of the different Delta Lake log operations listed in the mapped version entry.
```
Comment (N-o-Z) [inline docs/howto/hooks/lua.md:378]: Sentences are not very clear. table? list?JSON strings?? I'm not sure what I'm getting here

File: docs/howto/hooks/lua.md [hunk 4]
```diff
@@ -373,8 +373,17 @@ Creates a new Delta Lake client used to interact with the lakeFS server.
...
+- the second is a table of the metadata of the table, it consists of the following fields:
```
Comment (N-o-Z) [inline docs/howto/hooks/lua.md:379]: Isn't the metadata versioned as well?
I find it confusing that on the one hand we provide a list of versions for the table and on the other hand we provide only the metadata of the current snapshot. Perhaps a clarification is needed

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Overall looks very good!
I have some comments about the documentation.
Also - can we please add an esti test?

File: pkg/actions/lua/databricks/client.go (hunk 1 above)
Comment (N-o-Z) [reply pkg/actions/lua/databricks/client.go:12]: Don't mess with it, just remove the redundant newline 📏

Comment (Jonathan-Rosenberg) [reply pkg/actions/lua/databricks/client.go:12]: sure thing sure thing

File: docs/howto/hooks/lua.md (hunk 4 above)
Comment (Jonathan-Rosenberg) [reply docs/howto/hooks/lua.md:379]: It's the latest version's metadata.
The intention is to make it easier to get the current state of the table to initialize it in other systems faster.
I'll clarify

File: docs/howto/hooks/lua.md (hunk 3 above)
Comment (Jonathan-Rosenberg) [reply docs/howto/hooks/lua.md:378]: it's all true it's a lua [list](https://www.lua.org/pil/11.1.html), but all lists are a flavor of a table. The strings themselves are JSON strings

**Review by N-o-Z — APPROVED** [review]
Thanks for adding the tests!

**Pull request merged**